dev/provider_packages/refactor_provider_packages.py

```diff
@@ -146,6 +146,13 @@ def rename_deprecated_modules(self) -> None:
         for new, old in changes:
             self.qry.select_module(new).rename(old)
 
+        def is_not_k8spodop(node: LN, capture: Capture, filename: Filename) -> bool:
+            return not filename.endswith("/kubernetes_pod.py")
+
+        self.qry.select_module("airflow.providers.cncf.kubernetes.backcompat").filter(
+            callback=is_not_k8spodop
+        ).rename("airflow.kubernetes")
```

Comment:
Waaaaait -- just noticed that 1.10.x still has an `airflow/kubernetes/pod.py` etc. What is this going to do?

Reply:
Yeah. They should be removed after rename. However Interesting thing is that the package installed correctly (prepare backport providers job in CI does it).

Reply:
Yeah, pip doesn't really care if two distributions install the same file -- it'll just happily overwrite the existing file.

Which is fine right until you come to uninstall things.

This brings up an interesting point. What is the behaviour when upgrading from 1.10.x with backport provider to 2.0.0 with normal provider. Yes pip will complain about version conflicts, but that still isn't an error (especially given many people will be on an older pip still)

Reply:
From this respect good @ashb ! the backport provider package only contains airflow.providers files. We do not add airflow/kubernetes.

Just double-checked and installed both airflow 1.10.12 and backport cncf,kubernetes from pcakges and seem to be all ok.

This is what you get in 'backwards_compat_converters.py in the backport package:

```
from typing import List

from kubernetes.client import models as k8s

from airflow.exceptions import AirflowException
from airflow.kubernetes.pod import Port, Resources
from airflow.kubernetes.pod_runtime_info_env import PodRuntimeInfoEnv
from airflow.kubernetes.volume import Volume
from airflow.kubernetes.volume_mount import VolumeMount


def _convert_kube_model_object(obj, old_class, new_class):
    convert_op = getattr(obj, "to_k8s_client_obj", None)
    if callable(convert_op):
        return obj.to_k8s_client_obj()


```

Reply:
What is this `rename()` achieving then? What dist does/should it appear in?

Reply:
One small thing that we could do - maybe also remove the backcompat classes. They seem to not be removed during the refactoring and they are not needed in the backport at all:

![image](https://user-images.githubusercontent.com/595491/99374141-91370280-28c2-11eb-8bc8-53b85bbeb601.png)

Reply:
@ashb @dimberman Fixup here (I could not push it to astronomer's repo) : https://github.com/apache/airflow/pull/12392/commits/8d0be15ae4e4c1cd51d61ccd71fe0a8bef7f0ffe

Reply:
> This brings up an interesting point. What is the behaviour when upgrading from 1.10.x with backport provider to 2.0.0 with normal provider. Yes pip will complain about version conflicts, but that still isn't an error (especially given many people will be on an older pip still)

This will work fine. Backport packages need airflow ~1.10. PIP will not allow to upgrade airflow without removing backports.

Reply:
(I think that's cos Dan didn't tick the "allow commits from maintainers?" Or maybe that only works for PRs from personal repos -- I had the same trying to push to a PolideaInternal PR you opened)

Reply:
For example backport cncf.kubernetes:

```
       install_requires=[
            'apache-airflow>=1.10.12, <2.0.0',
            'cryptography>=2.0.0',
            'kubernetes>=3.0.0, <12.0.0',
        ],
```

Reply:
We should simply add a step in UPDATING to MD to make sure they remove backports simply (so that they are not tempted to force installation even with version conflict.

Reply:
> @ashb @dimberman Fixup here (I could not push it to astronomer's repo) : 8d0be15

Pushed this as https://github.com/apache/airflow/pull/12384/commits/d5d9e8b51b0ae534c8ac3f69a341c415ec8d3fb0

Reply:
@potiuk Turns out we need those files:

```
  File "/usr/local/lib/python3.6/site-packages/airflow/providers/cncf/kubernetes/operators/kubernetes_pod.py", line 39, in <module>
    from airflow.providers.cncf.kubernetes.backcompat.pod_runtime_info_env import PodRuntimeInfoEnv
ModuleNotFoundError: No module named 'airflow.providers.cncf.kubernetes.backcompat.pod_runtime_info_env'
```

Reply:
Force-pushed back to the previous commit.

Reply:
But we should not I believe. This import should be changed in the backport provider as well.

Reply:
So the proper fix will be to remove the classes from backcompat and add rename for that one as well.

Reply:
the kubernetes_pod.py was excluded from the previous rename, but pod_runtime_info can be changed in all classes.

Reply:
@dimberman ^^ have you seen this discussion ?